pep-0497-externals.txt

```diff
@@ -0,0 +1,359 @@
+number of package management tools, plus the changing nature of those, it
+was decided that concentrating on the specific files was more useful than
+using package names. It also provides looser coupling between the specification
+of externals and the package management tools.
+
+Location in setup.py rather than requirements.txt
+-------------------------------------------------
+
+Requirements.txt is scoped to the use for local deployments to specify
+requirements precisely for the relevant local environment. Setup.py is the
+appropriate place to specify the information for the package in general.
+
+In future, there may be some scope for including similar directives within
+requirements.txt, but that is out of scope for this PEP.
+
+Prior Approaches
+----------------
+
+Some capacity to support description of external dependencies (among other
+things) was described in PEPs 314 and 345 (Metadata 1.1 and 1.2). These
+describe directives whichs can be placed in the PKG-INFO file. An implementation
+responding to these directives is pending.
+
+This PEP places the directives in setup.py. This approach is closer to the
+typical approach of package maintainers to expressing requirements and places
+the external dependency specification nearer to the Python dependency
+specification. This PEP would supercede PEPs 314 and 345 with respect to
+external dependencies.
+
+Addition of vendor inference layer
+----------------------------------
+
+Bearing in mind the discussion about using specific dependencies, the ideal
+user experience is to get a useful message about how to get the package
+installed on their system. This means providing a message which is relevant
+for their package management environment (e.g. yum, apt, conda). Each
+distributor would be expected to provide an inference lookup which can be
+installed into pip which would assist with identifying the packages which
+are associated with the named dependencies.
+
+
+Types of environment and vendor
+-------------------------------
+
+Anaconda is not an operation system. Neither is yum, neither is apt. It is the
+maintainer of the packaging system which is pertinent, and the current operating
+system is an imperfect indicator of the relevant package manager.
+
+
+Directories, categories and locations of dependency files
+---------------------------------------------------------
+
+The proposed mechanism allows introspection of the standard location for
+packages. On *nix environments, these would be (for example), /usr/bin,
+/usr/local/include etc (see https://www.gnu.org/prep/standards/html_node/Directory-Variables.html
+for a discussion of standard directories). Non-standard dependency locations
+would be supported only via the underlying platform's mechanisms for exposing
+these details, such as PATH, LD_LIBRARY_PATH etc. This PEP does not include
+generalised configuration and specification of dependencies in non-standard
+or customised locations.
```

Comment:
I think we can say that pip etc should /honour/ those standard mechanisms but that its per-platform and not specified by this specification.